### ISSUE TITLE
Writing @HasQualifierParameter on a JDK class causes IllegalArgumentException

### ISSUE THREAD

**Issue by t-rasmud** [typetools/annotation-tools#269]
Adding the type qualifier `@HasQuailiferParameter` on `Collection` in the determinism checker
results in an `IllegalArgumentException` while building JDK.

[Bug report](https://travis-ci.org/t-rasmud/checker-framework/jobs/646613653#L14498).
```
Unknown error trying to insert annotations from: build/java/util/Collection.jaif to ../annotated/java/util/Collection.class
java.lang.IllegalArgumentException: value interface org.checkerframework.checker.determinism.qual.NonDet
	at org.objectweb.asmx.ClassWriter.newConstItem(ClassWriter.java:994)
	at org.objectweb.asmx.AnnotationWriter.visit(AnnotationWriter.java:185)
	at scenelib.annotations.io.classfile.ClassAnnotationSceneWriter.visitFields(ClassAnnotationSceneWriter.java:388)
	at scenelib.annotations.io.classfile.ClassAnnotationSceneWriter.ensureVisitSceneClassAnnotations(ClassAnnotationSceneWriter.java:245)
	at scenelib.annotations.io.classfile.ClassAnnotationSceneWriter.visitMethod(ClassAnnotationSceneWriter.java:190)
	at org.objectweb.asmx.ClassReader.accept(ClassReader.java:862)
	at org.objectweb.asmx.ClassReader.accept(ClassReader.java:445)
	at scenelib.annotations.io.classfile.ClassFileWriter.insert(ClassFileWriter.java:186)
	at scenelib.annotations.io.classfile.ClassFileWriter.main(ClassFileWriter.java:132)
```

Find the corresponding code changes [here](https://github.com/t-rasmud/checker-framework/compare/nondet-checker...t-rasmud:hasQualParamIssue).

Steps to reproduce the issue:

```
git clone git@github.com:t-rasmud/checker-framework.git -b hasQualParamIssue
cd checker-framework
export CHECKERFRAMEWORK=`pwd`
./gradlew cloneAndBuildDependencies
./gradlew buildJdk -PuseLocalJdk &> out
```

The command `./gradlew buildJdk -PuseLocalJdk &> out` should take about 4 mins. Grep for `IllegalArgumentExcepton` in `out.`

**Comment by mernst** [comment]
You gave no instructions for reproducing the problem.

Here is what I tried:

```
mkdir ~/tmp/issue269
cd ~/tmp/issue269
git clone git@github.com:t-rasmud/checker-framework.git -b hasQualParamIssue
cd checker-framework
export CHECKERFRAMEWORK=`pwd`
checker/bin-devel/build.sh
./gradlew -PuseLocalJdk buildJdk
```

Those commands succeeded (after 25 minutes).

Could you please give precise instructions to reproduce the problem?
It will be best if the test case is minimal, so that it runs quickly.

**Comment by t-rasmud** [comment]
@mernst I have updated the steps to reproduce the issue. I have also modified JDK's makefile to build only `determinism` JDK.

**Comment by mernst** [comment]
This may be fixed by b11894e0d98509110aa4cc3226b0029e0e5cbc43.
